harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -696,17 +694,23 @@ def _load(self, load_path: pathlib.Path) -> None:
 
         if "optimizer_state_dict" in checkpoint:
             # Backward compatible with older checkpoint format.
-            check.not_in("optimizers_state_dict", checkpoint)
-            check.eq(len(self.context.optimizers), 1)
+            if "optimizers_state_dict" in checkpoint:
+                raise RuntimeError(
+                    "Both optimizer_state_dict and optimizers_state_dict in checkpoint."
+                )
+            if len(self.context.optimizers) != 1:
+                raise RuntimeError()
```

Comment:
ditto

Reply:
Ditto, do you mean "Old-format checkpoint cannot be loaded into a **context** with more than one **optimizer**."?

Reply:
Done.